ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -444,45 +443,42 @@ private PlanNode buildSourceNode() {
           + analysis.getAllDataSources());
     }
 
-    return buildJoin((Join) tree);
+    return buildJoin((Join) tree, "");
   }
 
-
   /**
-   * @param root  the root of the Join Tree
+   * @param root    the root of the Join Tree
+   * @param prefix  the prefix to uniquely identify the plan node
    * @return the PlanNode representing this Join Tree
    */
-  private PlanNode buildJoin(final Join root) {
+  private PlanNode buildJoin(final Join root, final String prefix) {
```

Comment:
suggestion: Feels like this whole method would be better being somewhere outside of `LogicalPlanner`.

Reply:
I figured it would be easier to refactor it (cf your comment https://github.com/confluentinc/ksql/pull/5023#pullrequestreview-390659875) if everything was in one place, but if you want I can split this one out now

Reply:
np.